airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -938,7 +940,15 @@ void testSyncAfterUpgradeToPerStreamState(final TestInfo testInfo) throws Except
     assertEquals(1, state.getStreamState().size());
     final StreamState idAndNameState = state.getStreamState().get(0);
     assertEquals(new StreamDescriptor().namespace(PUBLIC).name(STREAM_NAME), idAndNameState.getStreamDescriptor());
-    assertEquals(Jsons.deserialize(expectedState), idAndNameState.getStreamState());
+    assertStateContains(Jsons.deserialize(expectedState), idAndNameState.getStreamState());
+  }
+
+  // Verify that actual contains expected all the (key, value) from expected.
+  private void assertStateContains(final JsonNode expected, final JsonNode actual) {
```

Comment:
I don't understand why we allow extra.

Reply:
https://github.com/airbytehq/airbyte/pull/15535 add new fields to the state, if possible, it would be nice to not have a hard check on the exact content of the state in the platform.